src/configurationProvider.ts

```diff
@@ -195,16 +194,115 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
+            return {
+                mainClass: config.mainClass,
+                projectName: config.projectName,
+            };
+        } else {
+            const errors: string[] = [];
+            if (!validationResponse.mainClass.passed) {
+                errors.push(String(validationResponse.mainClass.message));
+            }
+
+            if (!validationResponse.projectName.passed) {
+                errors.push(String(validationResponse.projectName.message));
+            }
+
+            if (validationResponse.proposals && validationResponse.proposals.length) {
+                const answer = await utility.showErrorMessageWithTroubleshooting({
+                    message: errors.join("\n"),
+                    type: Type.USAGEERROR,
+                }, "Quick Fix");
+                if (answer === "Quick Fix") {
+                    const quickFix: IMainClassOption = await this.showMainClassQuickPick(validationResponse.proposals,
+                        "Please select main class<project name>", true);
+                    this.persistLaunchConfig(folder, config, quickFix);
+                    return quickFix;
+                }
+            } else {
+                utility.showErrorMessageWithTroubleshooting({
+                    message: errors.join("\n"),
+                    type: Type.USAGEERROR,
+                });
+            }
+
+            return;
+        }
+    }
+
+    private persistLaunchConfig(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration, change: IMainClassOption): void {
+        if (!change) {
+            return;
+        }
+
+        const launchConfigurations: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("launch", folder);
+        const rawConfigs: vscode.DebugConfiguration[] = launchConfigurations.configurations;
+        const targetIndex: number = this.findDebugConfig(rawConfigs, config);
+        if (targetIndex >= 0) {
+            rawConfigs[targetIndex].mainClass = change.mainClass;
+            rawConfigs[targetIndex].projectName = change.projectName;
+
+            launchConfigurations.update("configurations", rawConfigs);
+        }
+    }
+
+    private findDebugConfig(rawConfigs: vscode.DebugConfiguration[], config: vscode.DebugConfiguration): number {
+        const candidates = [];
+        for (let i = 0; i < rawConfigs.length; i++) {
+            if (rawConfigs[i].name === config.name) {
+                candidates.push(i);
+            }
+        }
+
+        if (candidates.length > 1) {
+            for (const index of candidates) {
+                if (this.compareDebugConfig(rawConfigs[index], config)) {
+                    return index;
+                }
+            }
+
+            return candidates[0];
+        } else if (candidates.length === 1) {
+            return candidates[0];
+        }
+
+        return -1;
+    }
+
+    private compareDebugConfig(config1: object, config2: object): boolean {
```

Comment:
The convention of semantic for compare result is -1, 0, 1 but not boolean value. Please keep this convention as possible. 

Use equals*** for boolean result.

Reply:
this method is discarded.